mlrun/api/api/endpoints/grafana_proxy.py

```diff
@@ -0,0 +1,223 @@
+from http import HTTPStatus
+from typing import List, Dict, Any, Callable
+
+from fastapi import APIRouter, Response, Request
+
+from mlrun.api.api.endpoints.model_endpoints import get_access_key, list_endpoints
+from mlrun.api.schemas import ModelEndpointStateList, GrafanaTable, GrafanaColumn
+from mlrun.config import config
+from mlrun.utils.v3io_clients import get_frames_client
+
+router = APIRouter()
+
+
+def _parse_query_parameters(request_body: Dict[str, Any]) -> Dict[str, str]:
+    """
+    Looks for target field Grafana's SimpleJson query json body, parses semi-colon separated (;), key-value
+    queries.
+    """
+    targets = request_body.get("targets", [])
+    target_obj = targets[0] if targets else {}
+    target_query = target_obj.get("target") if target_obj else ""
+
+    if not target_query:
+        raise Exception(f"target missing in request body:\n {request_body}")
+
+    parameters = {}
+    for query in target_query.split(";"):
+        query_parts = query.split("=")
+        if len(query_parts) < 2:
+            raise Exception(
+                f"Query must contain both query key and query value. Expected query_key=query_value, "
+                f"found {query} instead."
+            )
+
+    return parameters
+
+
+def _validate_query_parameters(query_parameters: Dict[str, str]):
+    """Validates the parameters sent via Grafana's SimpleJson query"""
+    if "target_endpoint" not in query_parameters:
+        raise Exception(
+            f"Expected 'target_endpoint' field in query, found {query_parameters} instead"
+        )
+    if query_parameters["target_endpoint"] not in DISPATCH:
+        raise Exception(
+            f"{query_parameters['target_endpoint']} unsupported. Currently supported: {','.join(DISPATCH.keys())}"
+        )
+
+
+@router.get("/projects/grafana-proxy/model-endpoints", status_code=HTTPStatus.OK.value)
```

Comment:
`grafana-proxy` is not really a project
why doing it like that ?

Reply:
Any suggestions? Just mimicking the other endpoint related API. Grafana does things differently, it is based on the request body instead of the actual touting schema...